packages/devtools_app/lib/src/preferences.dart

```diff
@@ -13,10 +15,12 @@ class PreferencesController {
   final ValueNotifier<bool> _darkModeTheme = ValueNotifier(true);
   final ValueNotifier<bool> _vmDeveloperMode = ValueNotifier(false);
   final ValueNotifier<bool> _denseMode = ValueNotifier(false);
+  final ValueNotifier<String> _splitFractions = ValueNotifier('{}');
```

Comment:
It would be cleaner to have a separate
ValueNotifier<String> or better yet ValueNotifier<List<double>>. for each named split rather than one that returns a value that is JSON for all splits.
With the current approach, the size of the json blob will continue increasing as more splits are added to the app. There are only a couple splits on each page so not that many separate blobs will be loaded.

Reply:
A reasonable api would be 
```dart
ValueListenable<List<double> lookupSplitFractions(String key) {
  return _splitFractions.putIfAbsent(key, () { 
    final listenable = ValueListenable<List<double>>;
    // do work to listen on a preference with name 'splitters_$key' or similar
    // ....
    return listenable;
  })
}

Map<String, ValueListenable<List<double>>> _splitFractions;
```

Reply:
Actually never mind. Your current approach is probably better as it would be hard to make this approach sync.

Reply:
Ok here is an approach that would be cleaner but not require big changes.
```dart
ValueListenable<List<double> lookupSplitFractions(String key) {
  return _splitFractions.putIfAbsent(key, () { 
    final listenable = ValueListenable<List<double>>(_initialFractions[key]);
    return listenable;
  })
}

Map<String, ValueListenable<List<double>>> _splitFractions;
// Fill out _initialFractions in the init() method.
Map<String, List<double>> _initialFractions;
```